code/modules/mob/living/living.dm

```diff
@@ -1015,3 +1015,27 @@
 	if(mind && mind.linglink)
 		return LINGHIVE_LINK
 	return LINGHIVE_NONE
+
+/mob/living/proc/update_z(new_z) // 1+ to register, null to unregister
+	if (registered_z != new_z)
+		if (registered_z)
+			SSmobs.clients_by_zlevel[registered_z] -= src
+		if (client)
+			if (new_z)
+				SSmobs.clients_by_zlevel[new_z] += src
+			registered_z = new_z
+		else
+			registered_z = null
+
+/mob/living/Login()
```

Comment:
put this [here](https://github.com/tgstation/tgstation/blob/b9c54f517f7d62516624befcb168cb2ddae3b070/code/modules/mob/living/login.dm#L1)